source/Player.hx

```diff
@@ -187,6 +210,58 @@ import flixel.group.FlxGroup;
     btmBox.y = y + height - hitBoxHeight;
   }
 
+  //Override player.kill to drop any weilded items
+  override public function kill():Void 
+  {
+	  if (weilding){
+		dropCurrentEquip();
+	  }
+	  super.kill();
+	  new FlxTimer().start(2, resetLevel, 1); //This helps speed things up for debugging
+  }
+  
+  private function resetLevel(Timer:FlxTimer):Void
+	{
+		FlxG.resetState();
+	}
+
+	/**
+	 * Method to pickup items. If the player is not holding anything
+	 * and the item is weildable, equip the item. Otherwise add it to his bag.
+	 * If player is not weilding anything and the item is weildable
+	 *	set weilding to true and equip
+	*/
+	public function pickup_item(player:Player, item:Item):Void {
+		if (!weilding && item.weildable){
+			if(!item.justDropped){
+				weilding = true;
+				equipped_item = item;
+				item.equip(this);
+			}
+		} 
+	}
+
+	/**
+	 * Method to drop the currently equipped item
+	 * 
+	 */
+	private function dropCurrentEquip():Void{
+		equipped_item.drop_item();
+		weilding = false;
+		equipped_item = null;
+	}
+	
+	/**
+	 * Functionality for keypress event to attack if an item is currently being 
+	 * weilded. Disables user input to put them in the 'attacking state', but 
+	 * still keeps their current velocity and acceleration
+	 */
+	public function attack_state():Bool{
+		FlxG.keys.reset();
+		FlxG.keys.enabled = false;
```

Comment:
Not sure what happened, but one of my comments disappeared.  All the comments about locking input were supposed to be referring to this one.

This just feels bad to me.  It causes you to stop moving during your attack animation (you keep velocity, but not holding any movement key causes the player to come to a quick stop, and the `reset()` cancels any keys being held) and just makes it feel more clunky than I think is necessary.  If the intention here is just to prevent starting a new swing before the previous one is complete, I think just using a Bool flag and logic check in `Player` is a better choice than locking all input during the animation.